pages/docs/reference/collection-operations.md

```diff
@@ -0,0 +1,112 @@
+Results of such operations should be either stored in variables, or used in some other way, for example, passed in other functions.
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+fun main() {
+//sampleStart
+    val numbers = listOf("one", "two", "three", "four")  
+    numbers.filter { it.length > 3 }  // nothing happens with `numbers`, result is lost
+    val longerThan3 = numbers.filter { it.length > 3 } // result is stored in `longerThan3`
+//sampleEnd
+}
+
+```
+</div>
+
+For certain collection operations, there is an option to specify the _destination_ object.
+Destination is a mutable collection to which the function appends its resulting items instead of returning them in a new object.
+For performing operations with destinations, there are separate functions with the `To` postfix in their names, for example, [`filterTo()`](/api/latest/jvm/stdlib/kotlin.collections/filter-to.html) instead of [`filter()`](/api/latest/jvm/stdlib/kotlin.collections/filter.html)  or [`associateTo()`](/api/latest/jvm/stdlib/kotlin.collections/associate-to.html) instead of [`associate()`](/api/latest/jvm/stdlib/kotlin.collections/associate.html).
+These functions take the destination collection as an additional parameter.
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+fun main() {
+//sampleStart
+    val numbers = listOf("one", "two", "three", "four")
+    val filterResults = mutableListOf<String>()  //destination object
+    numbers.filterTo(filterResults) { it.length > 3 }
+    numbers.filterIndexedTo(filterResults) { index, _ -> index == 0 }
+    println(filterResults) // contains results of both operations
+//sampleEnd
+}
+```
+</div>
+
+For convenience, these functions return the destination collection back, so you can create it right in the corresponding argument of the function call:
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+fun main() {
+    val numbers = listOf("one", "two", "three", "four")
+//sampleStart
+    // filter numbers right into a new hash set, 
+    // thus eliminating duplicates in the result
+    val result = numbers.filterTo(HashSet()) { it.length > 3 }
+//sampleEnd
+}
+```
+</div>
+
+Functions with destination are available for filtering, association, grouping, flattening, and other operations. For the complete list of destination operations see the [Kotlin collections reference](/api/latest/jvm/stdlib/kotlin.collections/index.html).
+
+## Write operations
+
+For mutable collections, there are also _write operations_ that change the collection state. Such operations include adding, removing, and updating elements. Write operations are listed in the [Write operations](collection-write.html) and corresponding sections of [List specific operations](list-operations.html) and [Map specific operations](map-operations.html).
```

Comment:
Consider linking list- and map specific operations to the specific section anchors.